src/main/java/org/apache/commons/math4/stat/correlation/PearsonsCorrelation.java

```diff
@@ -159,13 +159,17 @@ public RealMatrix getCorrelationMatrix() {
      * @return matrix of correlation standard errors
      * @throws NullPointerException if this instance was created with no data
      */
+    @SuppressWarnings({"index:array.length.negative", "index:array.access.unsafe.high"}) /*
+    #1: nVars is @NonNegative as defined in #0.1
+    #2: i and j < nVars as checked by the loop conditions
+    */
     public RealMatrix getCorrelationStandardErrors() {
-        int nVars = correlationMatrix.getColumnDimension();
-        double[][] out = new double[nVars][nVars];
+        int nVars = correlationMatrix.getColumnDimension(); // #0.1
```

Comment:
Why does this need to be suppressed? Could you not just annotation `getColumnDimension` as `@NonNegative`?